src/libraries/System.Runtime.Loader/tests/AssemblyResolutionDowngradeTest.cs

```diff
@@ -0,0 +1,240 @@
+        public void CustomAssemblyLoadContextLoad_CanDowngradeVersion()
+        {
+            RemoteExecutor.Invoke(() => {
+                string assemblyV1Path = GetTestAssemblyPath("System.Runtime.Loader.Test.AssemblyVersion1");
+                string assemblyV3Path = GetTestAssemblyPath("System.Runtime.Loader.Test.AssemblyVersion3");
+                
+                var customContext = new DowngradeAssemblyLoadContext(assemblyV1Path);
+                
+                // Request version 3.0.0 but expect to get 1.0.0 via downgrade
+                var requestedAssemblyName = new AssemblyName($"{TestAssemblyName}, Version=3.0.0.0");
+                Assembly resolvedAssembly = customContext.LoadFromAssemblyName(requestedAssemblyName);
+                
+                Assert.NotNull(resolvedAssembly);
+                Assert.True(customContext.LoadCalled, "Custom Load method should have been called");
+                
+                // Verify we got the 1.0.0 assembly (downgrade successful)
+                Assert.Equal(new Version(1, 0, 0, 0), resolvedAssembly.GetName().Version);
+                
+                // Verify the assembly works as expected
+                Type testType = resolvedAssembly.GetType("System.Runtime.Loader.Tests.VersionTestClass");
+                Assert.NotNull(testType);
+                
+                string version = (string)testType.GetMethod("GetVersion").Invoke(null, null);
+                Assert.Equal("1.0.0", version);
+                
+                // Verify that the correct ALC loaded the assembly
+                Assert.Equal(customContext, AssemblyLoadContext.GetLoadContext(resolvedAssembly));
+            }).Dispose();
+        }
+
+        /// <summary>
+        /// Test that normal runtime resolution (without extension mechanisms) will NOT allow downgrades.
+        /// This test verifies the baseline behavior that downgrades only work via extension mechanisms.
+        /// </summary>
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void NormalResolution_CannotDowngradeVersion()
+        {
+            RemoteExecutor.Invoke(() => {
+                string assemblyV1Path = GetTestAssemblyPath("System.Runtime.Loader.Test.AssemblyVersion1");
+                
+                // First, load the version 1.0.0 assembly into the default context
+                Assembly loadedV1 = AssemblyLoadContext.Default.LoadFromAssemblyPath(assemblyV1Path);
+                Assert.Equal(new Version(1, 0, 0, 0), loadedV1.GetName().Version);
+                
+                // Now try to load version 3.0.0 - normal resolution should NOT automatically 
+                // downgrade to the already-loaded 1.0.0 version, it should fail
+                var requestedAssemblyName = new AssemblyName($"{TestAssemblyName}, Version=3.0.0.0");
+                
+                Assert.Throws<FileNotFoundException>(() => 
```

Comment:
When running on Mono, the downgrade actually succeeds. Update the test to account for this.